pkg/storage/stores/indexshipper/compactor/deletion/request_handler.go

```diff
@@ -74,14 +74,16 @@ func (dm *DeleteRequestHandler) AddDeleteRequestHandler(w http.ResponseWriter, r
 	}
 
 	deleteRequests := shardDeleteRequestsByInterval(startTime, endTime, query, userID, interval)
-	if _, err := dm.deleteRequestsStore.AddDeleteRequestGroup(ctx, deleteRequests); err != nil {
+	createdDeleteRequests, err := dm.deleteRequestsStore.AddDeleteRequestGroup(ctx, deleteRequests)
+	if err != nil {
 		level.Error(util_log.Logger).Log("msg", "error adding delete request to the store", "err", err)
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	level.Info(util_log.Logger).Log(
 		"msg", "delete request for user added",
+		"delete_request_id", createdDeleteRequests[0].RequestID,
```

Comment:
possible index-out-of-bounds error if the array is empty. not sure if it's possible now but `shardDeleteRequestsByInterval` might return an empty error for some reason(bug, wrong time range, etc...)

Reply:
I convinced myself that this could not happen but you provide a valid reason to add it. A test has been added.